File: geophone-dot-net/REST/GPResponse.cs
using System.Runtime.Serialization;

namespace geophone_dot_net.REST
{
    /// <summary>
    /// Response object for the GeoPhone REST API, containing provider and contact information,
    /// </summary>
    [DataContract]
    public class GPResponse 
    {
        public Provider[] Providers { get; set; }
        public Contact[] Contacts { get; set; }
        public Error Error { get; set; }
        public override string ToString()
        {
            string providers = Providers != null ? string.Join(", ", Providers.Select(p => p.ToString())) : "null";
            string contacts = Contacts != null ? string.Join(", ", Contacts.Select(c => c.ToString())) : "null";
            string error = Error != null ? Error.ToString() : "null";

            return $"GP GPResponse: Providers = {providers}, Contacts = {contacts}, Error = {error}";
        }
    }
    /// <summary>
    /// Represents a provider's information in the GeoPhone REST API response.
    /// </summary>  
    public class Provider
    {
        public string Name { get; set; }
        public string City { get; set; }
        public string State { get; set; }
        public string LineType { get; set; }
        public string Latitude { get; set; }
        public string Longitude { get; set; }
        public string Quality { get; set; }
        public override string ToString()
        {
            return $"ProviderOutput: Name = {Name}, City = {City}, State = {State}, LineType = {LineType}, Latitude = {Latitude}, Longitude = {Longitude}, Quality = {Quality}";
        }
    }

    /// <summary>
    /// Represents a contact's information in the GeoPhone REST API response.
    /// </summary>
    public class Contact
    {
        public string Name { get; set; }
        public string Address { get; set; }
        public string City { get; set; }
        public string State { get; set; }
        public string Zip { get; set; }
        public string Type { get; set; }
        public override string ToString()
        {
            return $"ContactOutput: Name = {Name}, Address = {Address}, City = {City}, State = {State}, Zip = {Zip}, Type = {Type}";
        }
    }

    public class Error
    {
        public string Desc { get; set; }
        public string Number { get; set; }
        public string Location { get; set; }
        public override string ToString()
        {
            return $"Desc: {Desc} " +
                $"Number: {Number} " +
                $"Location: {Location} ";
        }
    }

    public class LF_Provider
    {
        public string Name { get; set; }
        public string City { get; set; }
        public string State { get; set; }
        public string LineType { get; set; }

        public override string ToString()
        {
            return $"Name: {Name}, City: {City}, State: {State}, LineType: {LineType}";
        }
    }

    public class LF_Contact
    {
        public string Name { get; set; }
        public string Address { get; set; }
        public string City { get; set; }
        public string State { get; set; }
        public string Zip { get; set; }
        public string Type { get; set; }

        public override string ToString()
        {
            return $"Name: {Name}, Address: {Address}, City: {City}, State: {State}, Zip: {Zip}, Type: {Type}";
        }
    }

    public class LF_ResponseObject 
    {
        public Provider[] Providers { get; set; }
        public Contact[] Contacts { get; set; }
        public Error Error { get; set; }

        public override string ToString()
        {
            string providers = Providers?.Any() == true
                ? string.Join(", ", Providers.Select(p => p.ToString()))
                : "None";

            string contacts = Contacts?.Any() == true
                ? string.Join(", ", Contacts.Select(c => c.ToString()))
                : "None";

            return $"GP LF_Response: Providers: [{providers}], Contacts: [{contacts}], Error: {Error}";
        }
    }

}

File: geophone-nodejs/src/REST/gp-response.js
export class Provider {
    constructor(data) {
        this.Name = data.Name || '';
        this.City = data.City || '';
        this.State = data.State || '';
        this.LineType = data.LineType || '';
        this.Latitude = data.Latitude || '';
        this.Longitude = data.Longitude || '';
        this.Quality = data.Quality || '';
    }
    toString() {
        return `ProviderOutput: Name = ${this.Name}, City = ${this.City}, State = ${this.State}, LineType = ${this.LineType}, Latitude = ${this.Latitude}, Longitude = ${this.Longitude}, Quality = ${this.Quality}`;
    }
}

export class Contact {
    constructor(data) {
        this.Name = data.Name || '';
        this.Address = data.Address || '';
        this.City = data.City || '';
        this.State = data.State || '';
        this.Zip = data.Zip || '';
        this.Type = data.Type || '';
    }
    toString() {
        return `ContactOutput: Name = ${this.Name}, Address = ${this.Address}, City = ${this.City}, State = ${this.State}, Zip = ${this.Zip}, Type = ${this.Type}`;
    }
}

export class Error {
    constructor(data) {
        this.Desc = data.Desc || '';
        this.Number = data.Number || '';
        this.Location = data.Location || '';
    }
    toString() {
        return `Desc: ${this.Desc} Number: ${this.Number} Location: ${this.Location}`;
    }
}

export class GPResponse {
    constructor(data) {
        this.Providers = (data.Providers || []).map(p => new Provider(p));
        this.Contacts = (data.Contacts || []).map(c => new Contact(c));
        this.Error = data.Error ? new Error(data.Error) : null;
    }
    toString() {
        const providers = this.Providers.length ? this.Providers.map(p => p.toString()).join(', ') : 'null';
        const contacts = this.Contacts.length ? this.Contacts.map(c => c.toString()).join(', ') : 'null';
        const error = this.Error ? this.Error.toString() : 'null';
        return `GPResponse: Providers = ${providers}, Contacts = ${contacts}, Error = ${error}`;
    }
}
export default GPResponse;